### PR TITLE
Improve autofill APIs by moving dialog result handlers to autofill-impl

### PR DESCRIPTION
**Pull request by CDRussell**
Task/Issue URL: https://app.asana.com/0/488551667048375/1205232499768025/f 

### Description
Moves a lot of autofill result handling logic from `app` module into `autofill-impl` module, as per [Move AutofillCredentialsSelectionResultHandler to autofill-impl](https://app.asana.com/0/488551667048375/1205232499768025/f)

Some of the key highlights:
- Deleted `AutofillCredentialsSelectionResultHandler` which was doing too much and living in the `app` module. It is replaced with smaller, dedicated result handlers, each of which now live in `autofill-impl`
- Added `AutofillEventListener` to `autofill-api`, which  `BrowserTabFragment` implements. This is the main way events are communicated between autofill and the browser tab.
- Added `AutofillFragmentResultsPlugin` which allows a plugins-based approach for autofill-related fragments to delegate to the appropriate result handler living in `autofill-impl`.
- Moved `AutofillFireproofDialogSuppressor` from `app` module to `autofill-impl`

### Steps to test this PR

ℹ️ Since there is a large footprint of changes, a general smoke test of **credentials autofill** and **email protection** is needed.

**Saving, updating and autofilling login credentials**
- [x] Enable autofill from OVERFLOW -> Logins
- [x] Visit https://fill.dev/form/login-simple, and enter a username+password. Click Login button.
- [x] Verify you are prompted to save credential. 
- [x] Say "no"; verify it doesn't save
- [x] Repeat the above, and this time let it save. Verify it gets saved.
- [x] Revisit https://fill.dev/form/login-simple and verify you are prompted to use credential.
- [x] Say "no"; verify it doesn't autofill.
- [x] Repeat, and this time agree to let it autofill. Verify it correct autofills.
- [x] Revisit https://fill.dev/form/login-simple, and dismiss the prompt to autofill. Enter the same username you have saved but with a different password; verify it prompts to update the saved credential.

**Email protection**
- [x] Visit https://fill.dev/form/registration-email with email protection **disabled**
- [x] Verify no dax icon appears in the Email field
- [x] Enable email protection, and reload https://fill.dev/form/registration-email. Verify dax icon appears.
- [x] Tap on the dax icon and verify prompted to use email protection; test both buttons work as expected.

**Password generation**
- [x] Visit https://fill.dev/form/registration-email and verify icon shown in password fields
- [x] Tap on it, and verify prompted to generate a password. Test both options (agreeing and disagreeing) work as expected.

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -632,10 +632,6 @@
     <string name="autofillSnackbarAction" instruction="Label for a button that lets the user view the details for a login credential">View</string>
     <string name="autofillLoginSavedSnackbarMessage">Login saved</string>
     <string name="autofillLoginUpdatedSnackbarMessage">Login updated</string>
-    <string name="autofillDisableAutofillPromptTitle">Do you want to keep saving Logins?</string>
```
Comment (CDRussell) [inline app/src/main/res/values/strings.xml:635]: moved to `autofill-impl`

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -632,10 +632,6 @@
...
-    <string name="autofillDisableAutofillPromptMessage">You can disable this at any time in Settings.</string>
-    <string name="autofillDisableAutofillPromptPositiveButton">Keep Saving</string>
-    <string name="autofillDisableAutofillPromptNegativeButton">Disable</string>
```
Comment (CDRussell) [inline app/src/main/res/values/strings.xml:638]: Moved to `autofill-impl`

**Review by marcosholgado — COMMENTED** [review]
Haven't looked at the whole thing or tested it yet but some initial comments.

File: autofill/autofill-api/src/main/java/com/duckduckgo/autofill/api/AutofillEventListener.kt [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    /**
+     * Called when a login credential was updated. This API could be used to show visual confirmation to the user.
+     * @param credentials the login credential that were updated
+     */
+    fun onUpdatedCredentials(credentials: LoginCredentials)
+
+    /**
+     * Called when a change was detected in the autofill state, such that the current web page should be refreshed.
+     */
+    fun onRefreshRequested()
```
[the remark this answers is not in the document]
Comment (CDRussell) [reply autofill/autofill-api/src/main/java/com/duckduckgo/autofill/api/AutofillEventListener.kt:84]: Changed to:

```
fun onAutofillStateChange()
```

File: autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/email/ResultHandlerEmailProtectionChooseEmail.kt [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        autofillCallback: AutofillEventListener,
+    ) {
+        Timber.d("${this::class.java.simpleName}: processing result")
+
+        val userSelection: UseEmailResultType =
+            result.safeGetParcelable(EmailProtectionChooserDialog.KEY_RESULT) ?: return
+        val originalUrl = result.getString(EmailProtectionChooserDialog.KEY_URL) ?: return
+
+        when (userSelection) {
+            UsePersonalEmailAddress -> autofillCallback.onUseEmailProtectionPersonalAddress(originalUrl)
```
[the remark this answers is not in the document]
Comment (CDRussell) [reply autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/email/ResultHandlerEmailProtectionChooseEmail.kt]: Good suggestion, and I've made that change. We'd be able to move the pixels too after [API proposal: New plugin point to declare pixels requiring ATB and app version removal](https://app.asana.com/0/1202552961248957/1205522805984804/f)

**Review by marcosholgado — APPROVED** [review]
LGTM

**Pull request merged**